### ISSUE TITLE
How to exclude path with --android flag?

### ISSUE THREAD

**Issue by mrleolink** [pinterest/ktlint#602]
As the title says, how do I exclude some certain locations while using --android flag?

I have tried different ways like:

`ktlint '!*/build/generated/*' --android`
This apparently doesn't really scan anything

`ktlint '*.kt' '!*/build/generated/*' --android`
This doesn't respect the exclude pattern

`ktlint '*.kt' '!*/build/generated/*'`
This works but it doesn't apply android style.

---
Also, what do single asterisk and double asterisk mean in the pattern?
I tried both:
`ktlint '*.kt' '!**/build/generated/*'`
`ktlint '*.kt' '!*/build/generated/*'`
They gave same result.

**Comment by Tapchicoma** [comment]
You should have `--android` flag first:
```bash
$ ./ktlint --android '*.kt' '!*/build/generated/*'
```

**Comment by mrleolink** [comment]
Thanks for the suggestion @Tapchicoma , putting `--android` first doesn't help.

Yet I have figured out just by trial-and-error.

Either of this works:
`ktlint '*.kt' '!**/build/generated/*' --android`
`ktlint --android '*.kt' '!**/build/generated/*'`

The key is to use `**` for my negate pattern. I guess `**` matches any directory, and `*` matches any file.
That's just my guess, I would like to hear some confirmation from someone though.

**Comment by Tapchicoma** [comment]
Ktlint uses [klob](https://github.com/shyiko/klob) library for path matching, that implements `.gitignore` pattern style syntax: https://git-scm.com/docs/gitignore#_pattern_format

**Comment by mrleolink** [comment]
That explains everything.
Thanks
